sdk/go/pulumi/types.go

```diff
@@ -435,74 +450,6 @@ func AllWithContext(ctx context.Context, inputs ...interface{}) ArrayOutput {
 	return ToOutputWithContext(ctx, inputs).(ArrayOutput)
 }
 
-func gatherDependencies(v interface{}) []Resource {
-	if v == nil {
-		return nil
-	}
-
-	depSet := make(map[Resource]struct{})
-	gatherDependencySet(reflect.ValueOf(v), depSet)
-
-	if len(depSet) == 0 {
-		return nil
-	}
-
-	deps := make([]Resource, 0, len(depSet))
-	for d := range depSet {
-		deps = append(deps, d)
-	}
-	return deps
-}
-
-var resourceType = reflect.TypeOf((*Resource)(nil)).Elem()
-
-func gatherDependencySet(v reflect.Value, deps map[Resource]struct{}) {
```

Comment:
Still need to double check we're not missing any deps due to removing this code.

Reply:
Yeah, at a glance it seems like this would lose nested dependencies in an `Any`... why was it removed?

Reply:
I initially removed this to avoid gathering dependencies up-front in favor of forcing all dependencies to be awaited. I was trying to make it so that when creating an `Output`, you'd only pass a single up-front Resource as a dependency, and the only way to get the final set of all dependencies would be to await it.

`gatherDependencySet` was being called to gather up-front dependencies and it looked like the implementation of `awaitInputs` would have picked up most of the nested dependencies anyway, so that's why I initially removed it. However, `gatherDependencySet` does at least one thing that `awaitInputs` doesn't: if a nested value is a `Resource` the resource value is added to the set of dependencies.

In any case, I decided to add back `gatherDependencySet` to avoid missing the resource case. Now, you can continue to pass an array of up-front dependencies you know about when creating an `Output` (as is currently the case), and when you await it, you'll get a merged set of all dependencies (up-front + any other dependencies that came back as part of resolving the output).